src/fstests/runner.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Alexandra Pereira <alexandra.pereira@collabora.com>
+
+import os
+import sys
+
+import kernelci
+import kernelci.config
+import kernelci.db
+import kernelci.lab
+from kernelci.cli import Args, Command, parse_opts
+
+
+class FstestsRunner:
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+
+    def run(self):
+        sub_id = self._db.subscribe_node_channel({
+            'op': 'created',
+            'name': 'tarball',
+        })
+        print('Listening for tarballs')
+        print('Press Ctrl-C to stop.')
+        tarball_node_id = None
+        try:
+            while True:
+                if not tarball_node_id:
+                    tarball_node_id = self._db.receive_node(sub_id)
+                    print(f"New tarball from revision:\
+                         {tarball_node_id['revision']['commit'][:12]}")
+                    break
```

Comment:
Just fixing the logic here, if it's a loop it's a real loop:
```suggestion
        try:
            while True:
                tarball_node_id = self._db.receive_node(sub_id)
                print(f"New tarball from revision:\
                     {tarball_node_id['revision']['commit'][:12]}")
```

Reply:
Ah actually it't not `tarball_node_id` but `tarball_node` as it's the actual node data and not just the id.  The variable name is misleading.

Reply:
You could also print the tarball node id next to the git revision as that might be useful when debugging, in case the same git revision gets tested several times and new nodes get created for it.